test/micro/org/openjdk/bench/jdk/incubator/foreign/points/support/libJNIPoint.c

```diff
@@ -25,37 +25,45 @@
 
 #include "points.h"
 
+#ifdef _LP64
+#define JLONG_TO_PTR(value, type) ((type*) (value))
+#define PTR_TO_JLONG(value) ((jlong) (value))
+#else
+#define JLONG_TO_PTR(value, type) ((type*) (jint) (value))
```

Comment:
Maybe the jlong thisPoint argument comes from a pointer so it's ok.  Not nice, but if you say so, I'll go along.

Reply:
Yes, it's the same pointer that is returned from allocate. It's just stored in a `jlong` on the Java side (this would be a requirement on x64).

Do you have a suggestion for handling this otherwise?

Reply:
Hmm, now that I think about it, we could add overloads that work with `int` for 32-bit platforms.

But, for now 32-bit usage of these benchmarks seems unlikely. Since the build passes, and the benchmarks run on both platforms already, I'm reluctant to put much more effort into this at the moment.